File: maths/find_num_digits_in_number.cpp
#include <iostream>
#include <math.h>

int countDigitIter(long n)
{
    int count = 0;
    while (n != 0)
    {
        n = n/10;
        ++count;
    }
    return count;
}

int countDigitRecur(long n)
{
    if(n==0)
        return 0;
    return 1+countDigitRecur(n/10);
}

int countDigitLog(long n)
{
    return floor(log10(n)+1);
}

int main()
{
    int iter = countDigitIter(12345);
    int recur = countDigitRecur(12345);
    int logVal = countDigitLog(12345);
    stdout;
    std::cout << log10(12345);
    std::cout <<"Iter: "<< iter <<" Recur:" << recur <<" Logarthmic: " << logVal;
    return iter;
}

File: maths/find_num_digits_in_number.py
import math

def count_digits(num):
    print("log: ", math.floor(math.log10(num)+1))

def count_digits_recur(num):
    if num == 0:
        return 0
    return 1+count_digits_recur(num//10)

def count_digits_iter(num):
    count = 0
    while num != 0:
        num = num//10
        count += 1
    print("iteration: ", count)

count_digits(12345)
count_digits_iter(12345)
print("iteration: ", count_digits_recur(12345))